CHANGELOG.rst

```diff
@@ -5,32 +5,124 @@ Changelog
 
 0.14.0
 ------
-**Date:** TBA
+**Date:** 2021-06-XX
```

Comment:
Memo to myself: update me with final date

Reply:
## To Do

- [ ] docs: https://github.com/openPMD/openPMD-api/pull/1058#discussion_r674565930
- [ ] CI splitting & NVHPC: #1052
- [ ] Series throw file not found type: #1059